### PR TITLE
publish a bespoke codacy chart

### PR DESCRIPTION
**Pull request by joaocosta-codacy**
tldr:
feature: get latest versions of components from chartmuseum unstable
feature: publish bespoke chart with updated versions
feature: validate deployments after install

The aim of this job is to deploy to a self-hosted dev environment any artifact that which is published onto dev cloud.
This means that we could get early feedback on the artifacts we are producing, ahead of cutting a self-hosted release.
*First we fetch the latest versions of every component from the unstable channel on the chart museum.
*We also patch the worker image with the same name of this channel (unstable).
* Then we run an injected command to set the kubeconfig context to point to the right cluster
* We piggy back on helm history to know whether another installation is running and block until we can proceed
* We install the updated (local) chart
* Finally we validate the deployment of all components. If any component fails to get back up within 5m, the build will thus fail.

This is still in an early stage which means any feedback is very, very welcome :)

### CODE REVIEW

File: src/jobs/deploy_latest_codacy_chart.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+steps:
+  - run:
+      name: Checkout codacy chart
+      command: |
+        git clone https://github.com/codacy/chart.git
+  - run:
+      name: Patch artifact version and helm repo
+      command: |
+        pip3 install yq
+        DEPENDENCIES=$(yq .dependencies chart/codacy/requirements.yaml | jq .[].name |sed "s/\"//g" | grep -v "minio\|rabbitmq-ha\|postgresql\|log-router")
```
Comment (paulopontesm) [inline src/jobs/deploy_latest_codacy_chart.yaml]: can't we just use dymanic versions on requirements.yaml for this and make use of helm tools to do this?

the `helm update .` and `requirements.lock` are exacly for this purpose

Comment (paulopontesm) [reply src/jobs/deploy_latest_codacy_chart.yaml]: also  `grep -v "minio\|rabbitmq-ha\|postgresql\|log-router"` will create a lot of troubles when we decide to add new components. Again, the requirements.yaml should be used to say which versions are dynamic and which are not.

Comment (joaocosta-codacy) [reply src/jobs/deploy_latest_codacy_chart.yaml]: I have just addressed the grep part. 
We use jq to filter(select) only the components which have a git key in the requirements.yaml file. Those are the ones that belong to codacy and thus should be updated to the latest version.

Comment (joaocosta-codacy) [reply src/jobs/deploy_latest_codacy_chart.yaml]: The problem with dynamic versions is that some of our components publish as SNAPSHOTS, while others don't. 
For the dynamic version to pattern match against a SNAPSHOT, you must define the version as
^5.3.1-SNAPSHOT for instance. 
^5.3.1 will not match any snapshot :-(

Comment (andreaTP) [reply src/jobs/deploy_latest_codacy_chart.yaml]: this definitely looks like an hammer, I would rather go in the direction of uniforming the deployment of the components other than hammering here

**Comment by andreaTP** [comment]
I'm probably missing something, but can you elaborate on where(which CIs) you plan to use this orb command in?
The process for getting things in orbs is usually because of refactoring, is this code already used anywhere and you plan to use it more widely?

Last point is that this looks a lot of added complexity to maintain, e.g. Python installation of dependency etc. etc. we can probably think about having a docker image that statically contains all the requirement to avoid at least all the installations processes (that are slow) and possibly something like a "command line tool" to effectively perform the actions that we are doing here, WDYT @joaocosta-codacy ?

**Comment by joaocosta-codacy** [comment]
The purpose of this orb is to have a generic job that can be used at the end of the workflows of the components whenever they get published to dev. 
This would then bring us to the point where, whenever some component is published to dev cloud, it would also get published to dev self-hosted. It would allow us to react quicker to breaking changes in self-hosted well ahead of releases.
As it currently stands, this version of the orb is more for a proof of concept stage. All hammers will be refactored away. :)

**Comment by andreaTP** [comment]
@joaocosta-codacy would you mind to develop this as a PR to one component? and we refactor it back to orbs when we are done?
